src/include/btree_inline.h

```diff
@@ -2018,3 +2018,56 @@ __wt_bt_col_var_cursor_walk_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *
     cbt->cip_saved = cip;
     return (0);
 }
+
+/*
+ * __wt_btcur_skip_page --
+ *     Return if the cursor is pointing to a page with deleted records and can be skipped for cursor
+ *     traversal.
+ */
+static inline bool
+__wt_btcur_skip_page(WT_CURSOR_BTREE *cbt)
+{
+    WT_ADDR_COPY addr;
+    WT_PAGE *page;
+    WT_REF *ref;
+    WT_SESSION_IMPL *session;
+    uint8_t previous_state;
+    bool can_skip;
+
+    session = CUR2S(cbt);
+    ref = cbt->ref;
+    page = cbt->ref == NULL ? NULL : cbt->ref->page;
+
+    if (ref == NULL || page == NULL)
```

Comment:
Line 2039, the assignment for page takes care of ref being NULL. If ref is NULL, page will be NULL, so you can just check for page to be NULL here.

Reply:
Fixed.